src/common/stacks-utils.ts

```diff
@@ -60,16 +60,32 @@ export const stacksValue = ({
   const stxAmount = fixedDecimals
     ? parseFloat(stacks.toFormat(STX_DECIMALS))
     : stacks.decimalPlaces(STX_DECIMALS).toNumber();
-  return `${abbreviate && stxAmount > 10000 ? abbreviateNumber(stxAmount) : stxAmount}${
-    withTicker ? ' STX' : ''
-  }`;
+  return `${
+    abbreviate && stxAmount > 10000
+      ? abbreviateNumber(stxAmount)
+      : stxAmount.toLocaleString('en-US', {
+          maximumFractionDigits: fixedDecimals ? STX_DECIMALS : 3,
+        })
+  }${withTicker ? ' STX' : ''}`;
 };
 
 export const microStxToStx = (mStx: number | string) => {
   const microStacks = new BigNumber(mStx);
   return microStacks.shiftedBy(-STX_DECIMALS);
 };
 
+export const ftDecimals = (value: number | string, decimals: number) => {
+  const amount = new BigNumber(value);
+  return amount
+    .shiftedBy(-decimals)
+    .toNumber()
+    .toLocaleString('en-US', { maximumFractionDigits: decimals });
```

Comment:
`.toLocalString()` has difficulties past `MAX_SAFE_INTEGER`. Aren't there floating point imprecision issues here? I noticed them using this in the wallet. BigNumber has a toFormat method.